### DISCUSSION TITLE
How can i set a callback for case when permissions were allowed and dialog disappears?

### DISCUSSION THREAD

**Discussion by Gelembjuk** [jevonmao/PermissionsSwiftUI#103] [Q&A] [answered]
Hello All
I my app i want to detect when a user allows permissions i request and if allowed i want to execute some action 
I see there is onDisappear {} 
but as i understand it is executed in any case, not only when success fully accepted.

Also i see onDisappearHandler function with some 2 arguments, but it is undocumented.

What is the good way to do this?
maybe there is some way to check if permissions was already set for the app?

**Comment by jevonmao** [comment, marked as answer]
Hi there! Look into the `onDisappearHandler` callback for your use case. It is actually documented:
<img width="1114" alt="Screen Shot 2021-11-24 at 10 18 19 AM" src="https://user-images.githubusercontent.com/64660730/143293510-1c6cb5da-cca5-4800-9ac1-0eeb18793191.png">

You can use an if statement inside the callback to check whether the `authorizationStatus` of each of the `[JMResult]` is authorized. See definition of `JMResult` [here](https://github.com/jevonmao/PermissionsSwiftUI/blob/main/Sources/CorePermissionsSwiftUI/Model/Structs/JMResult.swift).
